Add more details about policy set execution mode (#1568)
hashicorp/web-unified-docs: content/terraform-docs-common/docs/cloud-docs/workspaces/settings/index.mdx

### Execution Mode

Determines the Terraform execution platform for this workspace. You can choose one of the following modes:

-  **Project Default**: HCP Terraform uses a project's [default execution mode](/terraform/cloud-docs/projects#execution-mode) to choose the execution platform for a workspace. This is the default setting.

- **Remote (custom)**: HCP Terraform performs Terraform runs on its own disposable virtual machines. This provides a consistent and reliable run environment and enables advanced features, such as Sentinel policy enforcement, cost estimation, notifications, and version control integration.

- **Local (custom)**: Lets you perform Terraform runs on the local workstation per the [CLI-driven run workflow](/terraform/cloud-docs/workspaces/run/cli). The workspace stores state, which Terraform can access with the [CLI integration](/terraform/cli/cloud). HCP Terraform does not evaluate workspace variables or variable sets in local execution mode. This mode disables remote execution.

> - **Agent (custom)**: Allows HCP Terraform to communicate with lightweight [HCP Terraform agents](/terraform/cloud-docs/agents). Use this mode to run Terraform in isolated, private, or on-premises infrastructure. When the policy execution mode is also set to **Agent**, HCP Terraform and Terraform Enterprise evaluate policies on an agent instance from the specified agent pool. Refer to [Configure connections to policy sets](/terraform/cloud-docs/workspaces/policy-enforcement/manage-policy-sets/configure) for more information.    

Reviewer: Trying to make this flow a little by keeping the link super specific:
```suggestion
- **Agent (custom)**: Lets HCP Terraform communicate with lightweight [HCP Terraform agents](/terraform/cloud-docs/agents). Use this mode to run Terraform in isolated, private, or on-premises infrastructure. If you also specify **Agent** as your [policy execution mode](/terraform/cloud-docs/workspaces/policy-enforcement/manage-policy-sets/configure#policy-set-exeuction-modes), HCP Terraform and Terraform Enterprise use the same agent instance to evaluate policies and run Terraform operations.
```
